commit 4b4a7c752ae305c168b0a3e89ad2f49c54cadd5d
Author: Hugo Pereira Da Costa <hugo.pereira@free.fr>
Date:   Thu Jun 12 14:56:20 2014 +0200

    Properly set titleHeight to zero when hideTitleBar is selected.
    CCBUG: 336128

diff --git a/kwin/clients/oxygen/oxygenclient.cpp b/kwin/clients/oxygen/oxygenclient.cpp
index a7d8d7a..ebc0274 100644
--- a/kwin/clients/oxygen/oxygenclient.cpp
+++ b/kwin/clients/oxygen/oxygenclient.cpp
@@ -394,8 +394,11 @@ namespace Oxygen
 
             case LM_TitleHeight:
             {
-                const int titleHeight = QFontMetrics(options()->font(true)).height();
-                return qMax(buttonSize, titleHeight);
+                if( hideTitleBar() ) return 0;
+                else {
+                    const int titleHeight = QFontMetrics(options()->font(true)).height();
+                    return qMax(buttonSize, titleHeight);
+                }
             }
 
             case LM_ButtonSpacing:
